### ISSUE TITLE
Passing lints to the lint function

### ISSUE THREAD

**Issue by csgillespie** [jimhester/lintr#351]
In the `lint()` help page, the arguement

```
linters | a list of linter functions to apply see linters for a full list of default and available linters.
```

This is correct. However, it needs to be a **named** list. So

```
lint("X.R", list(lint1 = T_and_F_symbol_linter))
```
is OK, but 
```
lint("X.R", list(T_and_F_symbol_linter))
```
this doesn't (and doesn't raise an error message).

**Comment by hfrick** [comment]
Hi @jimhester and @batpigandme, could we add this to the tidy-dev-day issues?

**Comment by batpigandme** [comment]
@jimhester would have to add the label! I don't have access to the repo! 😬
